flexmeasures/data/services/time_series.py

```diff
@@ -309,3 +310,43 @@ def set_bdf_source(bdf: tb.BeliefsDataFrame, source_name: str) -> tb.BeliefsData
     bdf = bdf.reset_index()
     bdf["source"] = DataSource(source_name)
     return bdf.set_index(index_cols)
+
+
+def drop_unchanged_beliefs(bdf: tb.BeliefsDataFrame) -> tb.BeliefsDataFrame:
+    """Drop beliefs that are already stored in the database with an earlier belief time.
+
+    Quite useful function to prevent cluttering up your database with beliefs that remain unchanged over time.
+    Only works on BeliefsDataFrames with a unique belief time and unique source.
+    """
+    if bdf.empty:
+        return bdf
+    if len(bdf.lineage.belief_times) > 1:
+        raise NotImplementedError("Beliefs should share a unique belief time.")
+    if len(bdf.lineage.sources) > 1:
+        raise NotImplementedError("Beliefs should share a unique source.")
+    previous_beliefs = bdf.sensor.search_beliefs(
```

Comment:
I'd call this `previous_beliefs_in_db` to be clear we're not searching in `bdf` (which a casual reader might see in this line without paying attention).